module_analysis/1_calc_module_composition.py

```diff
@@ -0,0 +1,283 @@
+
+# +
+pao1_annot_filename = paths.GENE_PAO1_ANNOT
+pa14_annot_filename = paths.GENE_PA14_ANNOT
+
+core_acc_dict = utils.get_my_core_acc_genes(
+    pao1_annot_filename, pa14_annot_filename, pao1_expression, pa14_expression
+)
+# -
+
+pao1_core = core_acc_dict["core_pao1"]
+pa14_core = core_acc_dict["core_pa14"]
+pao1_acc = core_acc_dict["acc_pao1"]
+pa14_acc = core_acc_dict["acc_pa14"]
+
+# ### Fisher's exact test
+
+# Get list of modules ids
+pao1_module_ids = pao1_membership["module id"].unique()
+pa14_module_ids = pa14_membership["module id"].unique()
+
+pao1_gene_group_composition = pd.DataFrame(
+    index=pao1_module_ids,
+    columns=["num core", "num acc", "odds ratio", "p-value", "module label"],
+)
+pa14_gene_group_composition = pd.DataFrame(
+    index=pa14_module_ids,
+    columns=["num core", "num acc", "odds ratio", "p-value", "module label"],
+)
+
+pao1_gene_group_composition.head()
+
+
+def label_modules(
+    module_ids_list, membership_df, core_genes_list, acc_genes_list, out_df
+):
+    all_genes = list(membership_df.index)
+
+    for module_id in module_ids_list:
+        # Find genes in module and outside module
+        genes_in_module = list(
+            membership_df[membership_df["module id"] == module_id].index
+        )
+        genes_outside_module = list(set(all_genes).difference(genes_in_module))
+
+        # Get core and accessory in module
+        core_genes_in_module = [
+            gene for gene in genes_in_module if gene in core_genes_list
+        ]
+        acc_genes_in_module = [
+            gene for gene in genes_in_module if gene in acc_genes_list
+        ]
+
+        # Get core and accessory genes outside of module
+        core_genes_outside_module = [
+            gene for gene in genes_outside_module if gene in core_genes_list
+        ]
+        acc_genes_outside_module = [
+            gene for gene in genes_outside_module if gene in acc_genes_list
+        ]
+
+        # Check
+        assert len(all_genes) == len(core_genes_in_module) + len(
+            acc_genes_in_module
+        ) + len(core_genes_outside_module) + len(acc_genes_outside_module)
+        assert len(genes_in_module) == len(core_genes_in_module) + len(
+            acc_genes_in_module
+        )
+
+        # Make contingency table
+        # -----|inside module|outside module
+        # core | # genes     | # genes
+        # acc  | # genes     | # genes
+        observed_contingency_table = np.array(
+            [
+                [len(core_genes_in_module), len(core_genes_outside_module)],
+                [len(acc_genes_in_module), len(acc_genes_outside_module)],
+            ]
+        )
+
+        # H0: The probability that the gene is core is the same
+        # whether or not you're in the module or outside
+        # H1: The probability that a gene is core is higher or lower inside the module
+        # than outside the module
+        # Should I do 2 one-sided tests, one for testing if
```

Comment:
Did you try the one-sided tests? Curious if it makes any difference.

Reply:
I did try running 1-sided and it didn't make a difference in the labeling